tensorboard/tools/diagnose_me.py

```diff
@@ -0,0 +1,382 @@
+          working with such names (https://bugs.python.org/issue26227).
+          Consider changing to a hostname that only contains printable
+          ASCII bytes.
+          """ % (binary_hostname,)
+      )
+      yield Suggestion("Use an ASCII hostname", message)
+    else:
+      message = reflow(
+          """
+          Python can't read your computer's hostname, %r. This can occur
+          if the hostname contains non-ASCII bytes
+          (https://bugs.python.org/issue26227). Consider changing your
+          hostname, rebooting your machine, and rerunning this diagnosis
+          script to see if the problem is resolved.
+          """ % (binary_hostname,)
+      )
+      yield Suggestion("Use a simpler hostname", message)
+    raise e
+
+
+@check
+def stat_tensorboardinfo():
+  # We don't use `manager._get_info_dir`, because (a) that requires
+  # TensorBoard, and (b) that creates the directory if it doesn't exist.
+  path = os.path.join(tempfile.gettempdir(), ".tensorboard-info")
+  logging.info("directory: %s", path)
+  try:
+    stat_result = os.stat(path)
+  except OSError as e:
+    if e.errno == errno.ENOENT:
+      # No problem; this is just fine.
+      logging.info(".tensorboard-info directory does not exist")
+      return
+    else:
+      raise
+  logging.info("os.stat(...): %r", stat_result)
+  logging.info("mode: 0o%o", stat_result.st_mode)
+  if stat_result.st_mode & 0o777 != 0o777:
+    preamble = reflow(
+        """
+        The ".tensorboard-info" directory was created by an old version
+        of TensorBoard, and its permissions are not set correctly; see
+        issue #2010. Change that directory to be world-accessible (may
+        require superuser privilege):
+        """
+    )
+    # This error should only appear on Unices, so it's okay to use
+    # Unix-specific utilities and shell syntax.
+    quote = getattr(shlex, "quote", None) or pipes.quote  # Python <3.3
+    command = "chmod 777 %s" % quote(path)
+    message = "%s\n\n\t%s" % (preamble, command)
+    yield Suggestion("Fix permissions on \"%s\"" % path, message)
+
+
+# Prefer to include this check last, as its output is long.
+@check
+def full_pip_freeze():
+  logging.info("pip freeze --all:\n%s", pip(["freeze", "--all"]).decode("utf-8"))
+
+
+def set_up_logging():
+  # Manually install handlers to prevent TensorFlow from stomping the
+  # default configuration if it's imported:
+  # https://github.com/tensorflow/tensorflow/issues/28147
+  logger = logging.getLogger()
+  logger.setLevel(logging.INFO)
+  handler = logging.StreamHandler(sys.stdout)
+  handler.setFormatter(logging.Formatter("%(levelname)s: %(message)s"))
+  logger.addHandler(handler)
+
+
+def main():
+  set_up_logging()
+
+  print("### Diagnostics")
+  print()
+
+  markdown_code_fence = "``````"  # seems likely to be sufficient
+  print(markdown_code_fence)
+  suggestions = []
+  for check in CHECKS:
+    print("--- check: %s" % check.__name__)
```

Comment:
Consider putting one newline after to give some space between the checks (alternatively, we can use the color but that complicates things I guess).

Reply:
Okay, if you prefer. (I like the concise form, but don’t feel strongly.)
I did consider color, but decided against it—color is easy on xterm
emulators, but this script needs to work on Windows, too.